### PR TITLE
[Feature branch] Mass delete of attributes

### PR DESCRIPTION
**Pull request by ghoussard**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Pim/Structure/Component/Repository/AttributeRepositoryInterface.php [hunk 1]
```diff
@@ -135,4 +135,8 @@ public function findAttributesByFamily(FamilyInterface $family);
      * @return array
      */
     public function findAvailableAxes($locale);
+
+    public function findByCodes(array $codes);
+
+    public function findByNotInCodes(array $codes);
```
Comment (mmetayer) [inline src/Akeneo/Pim/Structure/Component/Repository/AttributeRepositoryInterface.php:141]: missing phpdoc and typehinting, + it looks like these methods are not used anywhere, can you remove them?

File: tests/back/Acceptance/Attribute/InMemoryAttributeRepository.php [hunk 2]
```diff
@@ -252,4 +252,14 @@ public function findAvailableAxes($locale)
...
+
+    public function findByCodes(array $codes)
+    {
+        throw new NotImplementedException(__METHOD__);
+    }
+
+    public function findByNotInCodes(array $codes)
+    {
+        throw new NotImplementedException(__METHOD__);
+    }
```
Comment (mmetayer) [inline tests/back/Acceptance/Attribute/InMemoryAttributeRepository.php:264]: to remove?

File: tests/back/Pim/Structure/Integration/Attribute/Manager/AttributeCodeBlacklisterIntegration.php
Comment (mmetayer) [inline tests/back/Pim/Structure/Integration/Attribute/Manager/AttributeCodeBlacklisterIntegration.php:1]: would be cool to have tests with several attributes here

File: src/Akeneo/Pim/Structure/Bundle/Resources/config/datagrid/attribute.yml [hunk 3]
```diff
@@ -67,10 +67,6 @@ datagrid:
                 label: '%oro_datagrid.extension.orm_sorter.class%::DIRECTION_ASC'
         filters:
             columns:
-                code:
-                    type:      string
-                    label:     pim_common.code
-                    data_name: a.code
```
[the remark this answers is not in the document]
Comment (ValentinMumble) [reply src/Akeneo/Pim/Structure/Bundle/Resources/config/datagrid/attribute.yml:73]: Yes as the search bar search on both code & labels, it has been decided on a functional standpoint to remove this redundant filter (even though it allowed extra options like NOT IN)

File: src/Akeneo/Pim/Structure/Component/Query/InternalApi/CountAttributes.php
[the remark this answers is not in the document]
Comment (ValentinMumble) [reply src/Akeneo/Pim/Structure/Component/Query/InternalApi/CountAttributes.php:1]: Good catch, it's a leftover of a previous implem, I'll remove it 👍

File: src/Akeneo/Pim/Structure/Component/Repository/AttributeRepositoryInterface.php (hunk 1 above)
Comment (ValentinMumble) [reply src/Akeneo/Pim/Structure/Component/Repository/AttributeRepositoryInterface.php:141]: To remove as well

**Review by StevenVAIDIE — APPROVED** [review]

File: src/Akeneo/Pim/Structure/Bundle/Doctrine/ORM/Repository/AttributeRepository.php [hunk 4]
```diff
@@ -351,4 +351,27 @@ public function findAttributesByFamily(FamilyInterface $family)
...
+        $qb = $this->createQueryBuilder('attribute')
+            ->select('attribute');
+
+        if (!empty($codes)) {
+            $qb->where('attribute.code NOT IN (:codes)')
+                ->setParameter('codes', $codes);
+        }
+
+        return $qb->getQuery()->getResult();
+    }
```
Comment (mmetayer) [inline src/Akeneo/Pim/Structure/Bundle/Doctrine/ORM/Repository/AttributeRepository.php:376]: To remove as well

**Review by mmetayer — APPROVED** [review]
GTM once the last comment is resolved. Good job mates!

**Pull request merged**